csharp/src/Ice/IPEndpoint.cs

```diff
@@ -29,16 +34,21 @@ internal abstract class IPEndpoint : Endpoint
 
         internal const ushort DefaultIPPort = 4062;
 
+        /// <summary>Whether IPv6 sockets created from this endpoint are dual-mode or IPv6 only.</summary>
+        internal bool IPv6Only { get; }
+
         /// <summary>The source address of this IP endpoint.</summary>
         internal IPAddress? SourceAddress { get; }
 
+        internal bool IsOAEndpoint { get; }
```

Comment:
It's not clear to me why we need this property. And you're not using this property in Equals or GetHashCode.

Reply:
It's used in `AppendOptions`

Reply:
I don't think you need it there, like it was not needed for SourceAddress. If IsIPv6Only is false, you don't output anything; if IsIPv6Only is true, it can only be an OA endpoint.